core/store/src/db/rocksdb.rs

```diff
@@ -642,29 +645,35 @@ fn rocksdb_column_options(col: DBCol, store_config: &StoreConfig, temp: Temperat
         opts.set_compaction_filter("empty value filter", RocksDB::empty_value_compaction_filter);
     }
 
-    // Optimize write-heavy columns to boost compaction throughput
-    match col {
-        DBCol::PartialChunks | DBCol::State | DBCol::TrieChanges => {
-            opts.optimize_level_style_compaction(1024 * bytesize::MIB as usize);
-            opts.set_level_zero_file_num_compaction_trigger(8);
-            opts.set_level_zero_slowdown_writes_trigger(32);
-            opts.set_level_zero_stop_writes_trigger(64);
-            opts.set_max_subcompactions(2);
-            opts.set_target_file_size_base(128 * bytesize::MIB);
-            opts.set_max_write_buffer_number(8);
-            opts.set_compaction_readahead_size(6 * bytesize::MIB as usize);
-        }
-        DBCol::FlatState | DBCol::Chunks | DBCol::StateChanges | DBCol::Transactions => {
-            opts.optimize_level_style_compaction(512 * bytesize::MIB as usize);
-            opts.set_level_zero_file_num_compaction_trigger(8);
-            opts.set_level_zero_slowdown_writes_trigger(32);
-            opts.set_level_zero_stop_writes_trigger(64);
-            opts.set_target_file_size_base(128 * bytesize::MIB);
-            opts.set_max_write_buffer_number(6);
-            opts.set_compaction_readahead_size(4 * bytesize::MIB as usize);
-        }
-        _ => {}
+    // Column specific settings
+    let group_cfg = RocksDbCfConfig::resolve_for_column(col, &store_config.rocksdb);
+
+    // Apply group defaults derived from group_cfg
+    if let Some(v) = group_cfg.optimize_level_style_compaction {
+        opts.optimize_level_style_compaction(v.as_u64() as usize);
+    }
```

Comment:
I suggest destructuring `group_cfg` like this:

```rust
let RocksDbCfConfig { field1, field2, every_other_field.. } = RocksDbCfConfig::resolve_for_column(..);
```

and then handling each field that way. The reason to do is very simple: its easier this way for the compiler to tell us about fields we've missed applying here. It would automatically point out every location where these configs are applied whenever a new field is added (because the destructuring then becomes non-exhaustive.)

Same for `RocksDbConfig` above.

Reply:
Ah good point! It was still possible to add a new config and forget to use it in this file, this should fix the problem